docs/examples/snow-detection/snow-mode.py

```diff
@@ -0,0 +1,725 @@
+for v in dc_voltage_cols:
+    ax.scatter(data.index, data[v], s=0.5, label=v)
+    ax.plot(data[v], alpha=0.2)
+ax.axhline(mppt_high_voltage, c='r', ls='--',
+           label='Maximum MPPT voltage: {} V'.format(mppt_high_voltage))
+ax.axhline(mppt_low_voltage, c='g', ls='--',
+           label='Minimum MPPT voltage: {} V'.format(mppt_low_voltage))
+ax.set_xlabel('Date', fontsize='large')
+ax.set_ylabel('Voltage [V]', fontsize='large')
+ax.legend(loc='lower left')
+fig.tight_layout()
+plt.show()
+
+# %%
+# Plot AC power relative to inverter nameplate limits. We are looking for
+# periods of clipping, as these values are recorded outside of MPP operating
+# conditions.
+
+fig, ax = plt.subplots(figsize=(10, 6))
+date_form = DateFormatter("%m/%d")
+ax.xaxis.set_major_formatter(date_form)
+for a in ac_power_cols:
+    ax.scatter(data.index, data[a], s=0.5, label=a)
+    ax.plot(data[a], alpha=0.2)
+ax.axhline(max_ac_power, c='r', ls='--',
+           label='Maximum allowed AC power: {} kW'.format(max_ac_power))
+ax.set_xlabel('Date', fontsize='large')
+ax.set_ylabel('AC Power [kW]', fontsize='large')
+ax.legend(loc='upper left')
+fig.tight_layout()
+plt.show()
+
+# %% Filter data.
+# Identify periods where the system is operating off of its maximum power
+# point (MPP), and correct or mask. Conditions outside of the MPP cannot
+# be accurately modeled without external information on the system's
+# operating point. To allow us to make a valid comparison between system
+# measurements and modeled power at MPP, we set measurements collected below
+# the MPPT minimum voltage to zero, which emulates the condition where the
+# inverter turns off when it cannot meet the turn-on voltage.
+
+ac_power_cols_repeated = ac_power_cols + ac_power_cols + ac_power_cols
+for v, i, a in zip(dc_voltage_cols, dc_current_cols, ac_power_cols_repeated):
+
+    # Data where V < MPPT minimum
+    data.loc[data[v] < mppt_low_voltage, v] = 0
+    data.loc[data[v] < mppt_low_voltage, i] = 0
+    data.loc[data[v] < mppt_low_voltage, a] = 0
+
+    # Data where system is at Voc
+    data.loc[data[i] == 0, v] = 0
+
+# %% Plot DC voltage for each combiner input with inverter nameplate limits
+
+fig, ax = plt.subplots(figsize=(10, 10))
+date_form = DateFormatter("%m/%d")
+ax.xaxis.set_major_formatter(date_form)
+for v in dc_voltage_cols:
+    ax.scatter(data.index, data[v], s=0.5, label=v)
+    ax.plot(data[v], alpha=0.2)
+ax.axhline(mppt_high_voltage, c='r', ls='--',
+           label='Maximum MPPT voltage: {} V'.format(mppt_high_voltage))
+ax.axhline(mppt_low_voltage, c='g', ls='--',
+           label='Minimum MPPT voltage: {} V'.format(mppt_low_voltage))
+ax.set_xlabel('Date', fontsize='large')
+ax.set_ylabel('Voltage [V]', fontsize='large')
+ax.legend(loc='lower left')
+plt.show()
+
+# %%
+# We want to exclude periods where array voltage is affected by horizon
+# shading. Load in and apply horizon profiling created using approach described
+# in [1].
```

Comment:
The intent is to exclude times when the system output is reduced due to shading, rather than snow. I think it's optional in this example, and if some kind of exclusion is needed, we can replace with a solar zenith limit.

Reply:
I guess I don't understand why far shading would be conflated with snow losses anyway.  I haven't really thought it through, but assuming the horizon is shading both the array and the pyranometer (so observed and expected production are reduced in proportion), what's the problem?

Reply:
> Horizon shading is more or less equivalent to cloudy skies, 

Not in the way it is being used here. Clouds reduce irradiance across a fairly large area and so their effect is not to reduce string voltage but rather to reduce string current. Horizon shading can reduce voltage by casting shadows on only part of the array.

In any case the horizon masking is not necessary to illustrate how snow modes are classified.